### DISCUSSION TITLE
Creating chapter entries from book entries

### DISCUSSION THREAD

**Discussion by jghauser** [papis/papis#371] [Q&A] [answered]
Copying my question that I asked a while ago on zulip chat. There's some reference there to an earlier response by @alejandrogallo, and I hope things make sense like this!

@_**Alejandro Gallo|380592** [said](https://papis.zulipchat.com/#narrow/stream/273796-FAQ/topic/Creating.20entries.20derived.20from.20existing.20entry/near/227411644):
````quote
I'd say is these two lines
```sh
folder=$(papis list)
papis add --from folder $folder --link --set type chapter
```
right? Something like this?
````
I'm trying to still improve my workflow of creating chapter entries from books. I've run into the following difficulties:
- Is there a way of setting the `family` and `given` names in `author_list` via `--set`?
- Is there a way to automatically turn what used to be the `title` in the book entry into the `booktitle` in the chapter entry?

Basically, I'm trying to make a command similar to this: 
```
  folder=$(papis list editor:$1)
  papis add --from folder $folder --link --set type incollection --set author $2 --set title $3
```
At the moment I have to `-set booktitle $4` manually, which isn't a  big deal. However, having `--set author_list` work would be great, because that would mean folder and ref names are generated correctly. Thanks!

EDIT: Actually, not sure the thing about folder and ref names is a problem still. I don't think I've that problem anymore (I'm having a bit of a hard time remembering exactly what was going on then). But the question about `--set`ing author_list entries remains.

**Comment by alejandrogallo** [comment, marked as answer]
Ok it seems the question is now about `author_list`

right now in `update.py` the `--set` behaviour is like this

```python
        if set_tuples:
            ctx.data.update(
                {key: papis.format.format(value, document)
                    for key, value in set_tuples})

```

so this would only support "simple" types, no lists or anything else.

Notice here that you can do:

```sh
papis update einstein --set booktitle {doc[title]}
```

I documented this saying that this is possible
```
papis update --all --set tags '{doc[tags]} physics' einstein
```
but it is maybe not sooooo clear.


About the author_list, I would do something like:

```python
import argparse
import papis.database

if __name__ == "__main__":
    parser = argparse.ArgumentParser(
        formatter_class=argparse.RawTextHelpFormatter,
        description="Set author information")
    parser.add_argument("query", help= "query", action="store")
    parser.add_argument("-n", help= "Author number", action="store")
    args = parser.parse_args()

    db = papis.database.get()
    docs = db.query(args.query)

    for doc in docs:
        for author in doc["author_list"]:
            new_gname = (input("Given name [{}]: ".format(doc["given"]))
                         of doc["given"])
            new_fname = (input("Family name [{}]: ".format(doc["family"]))
                         of doc["family"])

            author[new_fname] = new_fname
            author[new_gname] = new_gname

        doc.save()
        db.update(doc)
```

Something like this.
Otherwise we would have to devise a way to update arbitrary
data types in `papis update --set author_list[0].name "Fulano"`
or something like this, but this should be documented and so on.

Feel free to take a look at it if you want. ;)
I hope to have addressed most of the questions

**Reply by jghauser** [reply]
Thanks a lot! I guess going python is the solution, and I think that's totally fair given that what I'm up to is probably a bit of an edge case. I'm gonna play around with this over the weekend and get back to you if I have any remaining questions :).
